platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -158,34 +64,36 @@ export function ViewportGridProvider({ children, service }) {
 
           const displaySetOptions = updatedViewport.displaySetOptions || [];
           if (!displaySetOptions.length) {
-            // Copy all the display set options, assuming a full set of displa
-            // set UID's is provided.
+            // Copy all the display set options, assuming a full set of displayset UID's is provided.
             displaySetOptions.push(...previousViewport.displaySetOptions);
             if (!displaySetOptions.length) {
               displaySetOptions.push({});
             }
           }
 
-          let newViewport = {
+          const newViewport = {
             ...previousViewport,
             displaySetInstanceUIDs,
             viewportOptions,
             displaySetOptions,
             viewportLabel: viewportLabels[viewportIndex],
           };
+
           viewportOptions.presentationIds = ViewportGridService.getPresentationIds(
             newViewport,
             viewports
           );
 
-          newViewport = reuseViewportId(
-            viewportIdSet,
-            newViewport,
-            state.viewports
-          );
+          if (!newViewport.viewportOptions?.viewportId) {
```

Comment:
In the if condition you handle an undefined viewportOptions, but then assign a value to it in the nested condition.  Either don't use the viewportOptions? or handle it if undefined.